src/azure-cli-core/azure/cli/core/commands/parameters.py

```diff
@@ -131,29 +131,31 @@ def enum_default(resource_type, enum_name, enum_val_name):
         return None
 
 
-def three_state_flag(positive_label='true', negative_label='false', invert=False):
+def three_state_flag(positive_label='true', negative_label='false', invert=False, return_label=False):
     """ Creates a flag-like argument that can also accept positive/negative values. This allows
     consistency between create commands that typically use flags and update commands that require
     positive/negative values without introducing breaking changes. Flag-like behavior always
     implies the affirmative unless invert=True then invert the logic.
     - positive_label: label for the positive value (ex: 'enabled')
     - negative_label: label for the negative value (ex: 'disabled')
     - invert: invert the boolean logic for the flag
+    - return_label: if true, return the corresponding label. Otherwise, return a boolean value
     """
     choices = [positive_label, negative_label]
 
     # pylint: disable=too-few-public-methods
     class ThreeStateAction(argparse.Action):
 
         def __call__(self, parser, namespace, values, option_string=None):
-            if invert:
-                if values:
-                    values = positive_label if values.lower() == negative_label else negative_label
-                else:
-                    values = values or negative_label
+            if not values:
+                values = negative_label if invert else positive_label
```

Comment:
From the description of `invert` it seems we would only invert the flag behavior (i.e. the flag implies false instead of true).  @devigned do you concur?